src/Graph.ts

```diff
@@ -117,6 +112,37 @@ export default class Graph {
 		this.phase = BuildPhase.GENERATE;
 	}
 
+	contextParse(code: string, options: Partial<acorn.Options> = {}) {
+		const onCommentOrig = options.onComment;
+		const comments: acorn.Comment[] = [];
+
+		if (onCommentOrig && typeof onCommentOrig == 'function') {
+			options.onComment = (block, text, start, end, ...args) => {
+				comments.push({type: block ? "Block" : "Line", value: text, start, end});
+				return onCommentOrig.call(options, block, text, start, end, ...args);
+			}
+		} else {
+			options.onComment = comments;
+		}
+
+		const ast = this.acornParser.parse(code, {
+			...(this.options.acorn as acorn.Options),
+			...options
+		});
+
+		if (typeof onCommentOrig == 'object') {
+			onCommentOrig.push(...comments);
+		}
+
+		options.onComment = onCommentOrig;
+
+		markPureCallExpressions(comments.map((cmt) => {
+			return {...cmt, block: cmt.type == "Block", text: cmt.value};
```

Comment:
This mapping strikes me as unnecessary if you either push the comments into the array in the previous way or change markPureCallExpressions to work with a slightly changed format?

Reply:
I agree. I will change it.

In general I try to keep my changes minimal because it's my first contribution and I don't want to accidentally break anything...